Directory.Build.targets

```diff
@@ -0,0 +1,8 @@
+<Project>
+  <ItemGroup>
+    <PackageReference Include="GitInfo" Version="2.0.20" PrivateAssets="All" />
+	  <PackageReference Include="MSBuilder.GenerateAssemblyInfo" Version="0.2.1" PrivateAssets="All" />
```

Comment:
the only projects that needs verioning are the Core, the Xaml, the Build.Tasks which are sdk-style. GenerateAssemblyIndo is not really necessary, so we take a dependency for very little value

Reply:
I m going to disagree, the goal of the PR is to add AssemblyInfo everywhere, and specially on all dlls that we ship on the nuget to be consistent with other products . 
When we move to sdk style projects be supported on vs4mac we can migrate all projects and drop this.